### PR TITLE
Add HART-IP protocol dissector

### PR DESCRIPTION
**Pull request by 0xA50C1A1**
Please sign (check) the below before submitting the Pull Request:

- [x] I have signed the ntop Contributor License Agreement at https://github.com/ntop/legal/blob/main/individual-contributor-licence-agreement.md
- [x] I have read the contributing guide lines at https://github.com/ntop/nDPI/blob/dev/CONTRIBUTING.md
- [x] I have updated the documentation (in doc/) to reflect the changes made (if applicable)

Describe changes:

HART-IP is an extension of the HART protocol, which is very popular in automation.

### CODE REVIEW

File: doc/protocols.rst [hunk 1]
```diff
@@ -216,3 +216,12 @@ References: `Wikipedia <https://en.wikipedia.org/wiki/Sina_Weibo>`_.
 IEEE 1588-2008 Precision Time Protocol (PTP) Version 2.
 
 References: `Protocol Specs: <https://standards.ieee.org/ieee/1588/4355/>`_.
+
+
+.. _Proto 358:
```
Comment (IvanNardi) [inline doc/protocols.rst]: Could you move it up, as entry 72 please? [see below]

File: src/include/ndpi_protocol_ids.h [hunk 2]
```diff
@@ -387,6 +387,7 @@ typedef enum {
   NDPI_PROTOCOL_SINA_WEIBO            = 356,
   NDPI_PROTOCOL_TESLA_SERVICES        = 357,
   NDPI_PROTOCOL_PTPV2                 = 358,
+  NDPI_PROTOCOL_HART_IP               = 359,
```
Comment (IvanNardi) [inline src/include/ndpi_protocol_ids.h]: Could you reuse `NDPI_PROTOCOL_FREE_72`, please?

File: src/lib/protocols/hart-ip.c [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+ */
+
+#include "ndpi_protocol_ids.h"
+
+#define NDPI_CURRENT_PROTO NDPI_PROTOCOL_HART_IP
+
+#include "ndpi_api.h"
+#include "ndpi_private.h"
+
+PACK_ON
```
Comment (IvanNardi) [inline src/lib/protocols/hart-ip.c]: Could you remove the two PACK directives, please? They should be useless, maybe even a little bit harmful

**Review by IvanNardi — CHANGES_REQUESTED** [review]

**Comment by 0xA50C1A1** [comment]
@IvanNardi Thanks for the review, I've fixed it.

File: doc/protocols.rst [hunk 4]
```diff
@@ -216,3 +216,12 @@ References: `Wikipedia <https://en.wikipedia.org/wiki/Sina_Weibo>`_.
...
 References: `Protocol Specs: <https://standards.ieee.org/ieee/1588/4355/>`_.
+
+
+.. _Proto 72:
+
+`NDPI_PROTOCOL_HART_IP`
+============================
+Highway Addressable Remote Transducer over IP
+
+References: `Protocol Specs: <https://library.fieldcommgroup.org/20085/TS20085>`_.
```
Comment (IvanNardi) [inline doc/protocols.rst]: Could you move this new section up, keeping the protocol numbers ordered? 72/HART should be the first of the list....

Comment (0xA50C1A1) [reply doc/protocols.rst]: Done

**Review by IvanNardi — APPROVED** [review]

**Comment by IvanNardi** [comment]
Thanks!

**Pull request merged**